.github/workflows/build.yml

```diff
@@ -4,12 +4,12 @@ jobs:
   build:
     strategy:
       matrix:
-        java: [8-jdk, 11-jdk, 17-jdk, 21-jdk]
+        java: [17-jdk, 21-jdk]
     runs-on: ubuntu-22.04
     container:
       image: eclipse-temurin:${{ matrix.java }}
       options: --user root
     steps:
       - uses: actions/checkout@v1
       - uses: gradle/wrapper-validation-action@v1
-      - run: ./gradlew build --stacktrace --warning-mode fail
+      - run: ./gradlew check --stacktrace --warning-mode fail
```

Comment:
This is wrong, build depends on check. With this the code wont be compiled or tested.

Reply:
Oh didn't know that. `check` depends on compile and test though.

Reply:
Ah, seems it does run the tests. build is still want we want though as that will make sure everything is ran.

Reply:
`check` depends on `test` in particular, which also needs to compile the main code since it's on the test classpath - but `build` is better since it also tests creating the jars (the `assemble` task)